### DISCUSSION TITLE
Validation entre deux dates

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#216] [Annonces]
Bonjour,

J'ai le code suivant pour valider 2 date, qu'une soit plus grand que l'autre :

type: inline
              components:                   
              - type: date
                name: DSpecProg_dateDebutPrevProj
                label:
                  fr: Date de début prévue du projet

              - type: date
                name: DSpecProg_dateFinPrevProj
                label:
                  fr: Date de fin prévue du projet
                validations:
                  dateFinPlusGrandEgalDateDebutCustom:
                additionals:
                  :validation-rules: |
                    {
                      dateFinPlusGrandEgalDateDebutCustom: ({ value }) => {
                        return value >= DSpecProg_dateDebutPrevProj;
                      }                          
                    }               
                validation-messages:
                  dateFinPlusGrandEgalDateDebutCustom: 
                    fr: La date de fin prévue du projet doit être supérieure ou égale à la date de début prévue du projet.

La validation ne décolle pas et en plus j'ai toujours le message que la date de fin est obligatoire même s'il y a une date. Il y a quelque chose qui m'échappe.

**Comment by j0ber** [comment]
Il faut débugger votre javascript... dans votre cas sans essayer je peux déjà dire que
return value >= DSpecProgdateDebutPrevProj    ne fonctionnera pas...

Qu'est-ce que c'est "DSpecProgdateDebutPrevProj " ?

Si c'est un champ ce n'est pas de cette façon que vous pouvez obtenir sa valeur tel que défini dans notre documentation. Voir https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=sectionInjecterJs regarder dans la section des concepts de base. On explique comment obtenir la valeur d'un champ...  

C'est dans la même page où je vous ai ajouté un exmple de "faux tableau" récemment...

**Comment by ChrisThe01** [comment]
Merci ! Pourtant je me suis inspiré des exemples dans la section "Validation personnalisées". Ok je vais revoir ma solution...

**Reply by j0ber** [reply]
Oui tout semble bon... sauf la façon d'obtenir la valeur du champ avec lequel vous voulez comparer votre date. Il ne vous manque que ce qu'il faut pour obtenir la valeur de DSpecProg_dateDebutPrevProj. C'est presque rien. Tout est dans ma réponse initiale.
